os/include/semaphore.h

```diff
@@ -106,18 +109,28 @@ struct semholder_s {
 #endif
 #endif							/* CONFIG_PRIORITY_INHERITANCE */
 
+#ifdef CONFIG_APP_BINARY_SEPARATION
+#define IS_KERNEL_SEM(sem) ((sem->flags & FLAGS_KERNELSEM) != 0)
```

Comment:
One question, is named semaphore also included in kernel sem?

Reply:
Right, it is created in kernel space.